akka-actor/src/main/scala/akka/actor/AbstractActor.scala

```diff
@@ -46,29 +50,97 @@ object AbstractActor {
  */
 abstract class AbstractActor extends Actor {
 
-  private var _receive: Receive = null
-
-  /**
-   * Set up the initial receive behavior of the Actor.
-   *
-   * @param receive  The receive behavior.
-   */
-  @throws(classOf[IllegalActorStateException])
-  protected def receive(receive: Receive): Unit =
-    if (_receive == null) _receive = receive
-    else throw IllegalActorStateException("Actor behavior has already been set with receive(...), " +
-      "use context().become(...) to change it later")
-
   /**
    * Returns this AbstractActor's AbstractActorContext
    * The AbstractActorContext is not thread safe so do not expose it outside of the
    * AbstractActor.
    */
   def getContext(): AbstractActorContext = context.asInstanceOf[AbstractActorContext]
 
-  override def receive =
-    if (_receive != null) _receive
-    else throw IllegalActorStateException("Actor behavior has not been set with receive(...)")
+  /**
+   * Returns the ActorRef for this actor.
+   */
+  def getSelf(): ActorRef = self
```

Comment:
So this is mostly for source compatibility with UntypedActor.
Could we deprecate them right away then? Since it would be nice to have 1 API for those, it should be the `sender()` and `self()` ones.